parley/src/layout/editor.rs

```diff
@@ -15,12 +15,27 @@ use alloc::{sync::Arc, vec::Vec};
 
 #[derive(Copy, Clone, Debug)]
 pub enum ActiveText<'a> {
-    /// The selection is empty and the cursor is a caret; this is the text of the cluster it is on
+    /// The selection is empty and the cursor is a caret; this is the text of the cluster it is on.
     FocusedCluster(Affinity, &'a str),
-    /// The selection contains this text
+    /// The selection contains this text.
     Selection(&'a str),
 }
 
+/// Opaque representation of a generation.
+// Overflow handling: the generations are only compared,
+// so wrapping is usually fine, except that in the unlikely
+// event that exactly `u32::MAX` generations happen before the
+// application gets around to drawing `Generation(0)`.
+#[derive(PartialEq, Default, Clone, Copy)]
+pub struct Generation(u32);
```

Comment:
A minor detail, but maybe users should not be able to construct `Generation` themselves. OTOH, that makes constructing wrapping structs a bit more cumbersome (e.g., for vello_editor, `PlainEditor` then has to be assigned to a variable before constructing `SimpleVelloApp`). At the same time, currently the editor's first generation and the default generation are 0, that may lead to bugs in user code.

Users can get easy constructing again by using `last_generation: Option<Generation>`, but then generation should ideally wrap [NonZeroU32](https://doc.rust-lang.org/stable/std/num/type.NonZeroU32.html).

Reply:
I kind of like this. But comparing `Option<Generation>` and `Generation` is kind of ugh-ly.